framework/include/distributions/Distribution.h

```diff
@@ -0,0 +1,45 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef DISTRIBUTION_H
+#define DISTRIBUTION_H
+
+#include "MooseObject.h"
+#include "MooseRandom.h"
+
+class Distribution;
+
+template<>
+InputParameters validParams<Distribution>();
+
+class Distribution :
+  public MooseObject
+{
+public:
+  Distribution(const InputParameters & parameters);
+  virtual ~Distribution();
+
+  virtual Real pdf(const Real & x) = 0;
+  virtual Real cdf(const Real & x) = 0;
+  virtual Real inverseCdf(const Real & y) = 0;
+  /// random sample
+  virtual Real random();
+  unsigned int getSeed();
+  virtual void setSeed(unsigned int seed);
+
+  unsigned int _seed;
+  MooseRandom _random;
+};
+
+#endif /* DISTRIBUTION_H */
```

Comment:
This entire class needs to be documented. Methods use `/** */` style and members use `///`.